lib/bignum.toit

```diff
@@ -0,0 +1,86 @@
+// Copyright (C) 2022 Toitware ApS. All rights reserved.
+// Use of this source code is governed by an MIT-style license that can be
+// found in the lib/LICENSE file.
+
+import encoding.hex
+
+BIGNUM_ADD_ ::= 0
+BIGNUM_SUB_ ::= 1
+BIGNUM_MUL_ ::= 2
+BIGNUM_DIV_ ::= 3
+BIGNUM_MOD_ ::= 4
+
+class Bignum:
+  sign_/bool := ?
+  limbs_/ByteArray := ?
+
+  constructor .sign_/bool .limbs_/ByteArray:
+
+  constructor.from_string data/string:
+    if data[0] == '-':
+      sign_ = true
+      data = data[1..]
+    else:
+      sign_ = false
+    
+    if data.size & 0x1 !=0:
+      data = "0" + data
+    
+    limbs_ = hex.decode data
+
+  int_to_arry_ i/int -> ByteArray:
+    return #[(i >>  0) & 0xff, (i >>  8) & 0xff,
+             (i >> 16) & 0xff, (i >> 24) & 0xff]
+  
+  trans_other_ other/any -> Bignum:
+    if other is int:
+      other = Bignum
+          other < 0
+          int_to_arry_ other.abs
+    else if other is not Bignum:
+      throw "$(other.type) is not supported"
+    
+    return other
+
+  basic_operator_ operation/int sign/bool limbs/ByteArray other/any:
```

Comment:
Since this method is not static it doesn't need sign and limbs as arguments, it can get them from `this`.